tests/utils/docker.spec.ts

```diff
@@ -38,8 +38,19 @@ describe('getDefaultDockerModemOpts() function', function () {
 			host: undefined,
 			port: undefined,
 			protocol: 'http',
-			socketPath: defaultSocketPath,
 		});
+		if (typeof defaultOps.socketPath === 'function') {
+			// Function is always findDefaultUnixSocket(), which returns a promise.
+			// Must override type since @types/dockerode not updated yet.
+			const socketPath: () => Promise<string> = defaultOps.socketPath;
+			socketPath()
+				.then((socketPathRes) =>
+					expect(socketPathRes).to.equal(defaultSocketPath),
+				)
+				.catch((e) => assert.fail(`socketPath() failed: ${e}`));
```

Comment:
I don't think this will work as you expect since it's in a promise that is neither awaited nor returned, I would simply rewrite this with async/await to avoid the issue:
```suggestion
			try {
				expect(await socketPath()).to.equal(defaultSocketPath),
			} catch (e) {
				throw new Error(`socketPath() failed: ${e}`));
			}
```
I also switched `assert.fail` to `throw new Error` as it should be equivalent and avoids the need for the extra import. I would also remove the `catch` block altogether if the custom error message isn't necessary but that doesn't matter to me either way

Reply:
I agree, much simpler. Also removed catch block.